components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/DefaultResponseBuilder.java

```diff
@@ -42,10 +48,34 @@
     public Response buildResponse(SAMLSSOAuthnReqDTO authReqDTO, String sessionId)
             throws IdentityException {
 
+        DateTime issueInstant = new DateTime();
+        return buildResponse(authReqDTO, sessionId, issueInstant, null);
+    }
+
+    @Override
+    public Response buildResponse(SAMLSSOAuthnReqDTO authReqDTO, String sessionId, DateTime issueInstant,
+                                  String assertionId) throws IdentityException {
+
         if (log.isDebugEnabled()) {
             log.debug("Building SAML Response for the consumer '"
                     + authReqDTO.getAssertionConsumerURL() + "'");
         }
+
+        Assertion assertion;
+        if (authReqDTO.isAssertionQueryRequestProfileEnabled() && assertionId != null) {
+            SAML2ArtifactInfoDAO saml2ArtifactInfoDAO = new SAML2ArtifactInfoDAOImpl();
+            try {
+                assertion = saml2ArtifactInfoDAO.getSAMLAssertion(assertionId);
+            } catch (ArtifactBindingException e) {
+                throw new IdentityException("Error while retrieving SAML assertion from the database.", e);
```

Comment:
throw new IdentityException("Error while retrieving SAML assertion from the database. assertionId : "+ assertionId, e);